docs/acceptance-criteria/uacs-tbtc-v2.adoc

```diff
@@ -0,0 +1,291 @@
+:toc: macro
+
+= tBTCv2 User Acceptance Criteria
+
+:icons: font
+:numbered:
+toc::[]
+
+== Purpose of the document
+This document stores User Acceptance Criteria for TBTCv2. It will be used to
+gain a common understanding of how different components of TBTCv2 should act and
+will be utilized during the acceptance testing of the feature.
+
+== User Acceptance Criteria list
+
+[%header,cols="1,4"]
+|===
+| Area
+| UAC
+
+| Wallet lifecycle
+| Anyone can send a notification to the wallet registry contract informing that
+  it's time to create a new wallet.
+
+// TODO: Add UAC describing ECDSA client perspective - clients should be
+// programmed to send the notifications when they notice that new wallet
+// creation criteria are met. Do we know how to discourage sending the
+// notifications by multiple clients at the same time?
+
+| Wallet lifecycle - wallet creation
+| Upon processing of the time-to-create-new-wallet notification, the wallet
+  registry contract should determine the validity of the notification.
+
+| Wallet lifecycle - wallet creation
+| When determining the validity of the time-to-create-new-wallet notification,
+  the following aspects should be taken into account: +
+  - is the creation of a new wallet already in progress? (if yes, then
+  notification invalid) +
+  - have at least `wallet_creation_period` blocks passed since the last wallet
+  was created? (if not, then notification invalid) +
+  - are there no wallets or do all wallets have their balances higher or equal
```

Comment:
Should we take under consideration the state of the wallets (for example, if the wallet is in the 'emptying' phase, should that change anything or should it be treated as a regular wallet)?